components/locale_core/src/shortvec/litemap.rs

```diff
@@ -83,13 +92,39 @@ impl<K, V> StoreMut<K, V> for ShortBoxSlice<(K, V)> {
     fn lm_clear(&mut self) {
         self.clear();
     }
+}
 
+#[cfg(feature = "alloc")]
+impl<K: Ord, V> StoreBulkMut<K, V> for ShortBoxSlice<(K, V)> {
     fn lm_retain<F>(&mut self, mut predicate: F)
     where
         F: FnMut(&K, &V) -> bool,
     {
         self.retain(|(k, v)| predicate(k, v))
     }
+
+    fn lm_extend<I>(&mut self, other: I)
+    where
+        I: IntoIterator<Item = (K, V)>,
+    {
+        let other = other.into_iter();
+        let mut items = alloc::vec::Vec::with_capacity(self.len() + other.size_hint().0);
```

Comment:
nit: this needlessly allocates in some cases (self is empty, iterator is size 1 or 0, or self is size 1 and iterator is size 0)